`./configure $CONFIGURE_ARGS` leads to linker errors on Ubuntu 20.04

Whenever I try `./configure $CONFIGURE_ARGS` instead of `./configure` as recommended by the README, I get

```
/home/sgraf-local/code/hs/ghc/misc2/_validate/stage1/lib/../lib/x86_64-linux-ghc-9.3.20211124/libHSghc-bignum-1.3-ghc9.3.20211124.so: error: undefined reference to '__gmpn_lshift'
/home/sgraf-local/code/hs/ghc/misc2/_validate/stage1/lib/../lib/x86_64-linux-ghc-9.3.20211124/libHSghc-bignum-1.3-ghc9.3.20211124.so: error: undefined reference to '__gmpn_rshift'
/home/sgraf-local/code/hs/ghc/misc2/_validate/stage1/lib/../lib/x86_64-linux-ghc-9.3.20211124/libHSghc-bignum-1.3-ghc9.3.20211124.so: error: undefined reference to '__gmpz_get_d'
/home/sgraf-local/code/hs/ghc/misc2/_validate/stage1/lib/../lib/x86_64-linux-ghc-9.3.20211124/libHSghc-bignum-1.3-ghc9.3.20211124.so: error: undefined reference to '__gmpz_get_d_2exp'
/home/sgraf-local/code/hs/ghc/misc2/_validate/stage1/lib/../lib/x86_64-linux-ghc-9.3.20211124/libHSghc-bignum-1.3-ghc9.3.20211124.so: error: undefined reference to '__gmpn_gcd_1'
/home/sgraf-local/code/hs/ghc/misc2/_validate/stage1/lib/../lib/x86_64-linux-ghc-9.3.20211124/libHSghc-bignum-1.3-ghc9.3.20211124.so: error: undefined reference to '__gmpz_init'
/home/sgraf-local/code/hs/ghc/misc2/_validate/stage1/lib/../lib/x86_64-linux-ghc-9.3.20211124/libHSghc-bignum-1.3-ghc9.3.20211124.so: error: undefined reference to '__gmpz_gcd'
/home/sgraf-local/code/hs/ghc/misc2/_validate/stage1/lib/../lib/x86_64-linux-ghc-9.3.20211124/libHSghc-bignum-1.3-ghc9.3.20211124.so: error: undefined reference to '__gmpz_clear'
/home/sgraf-local/code/hs/ghc/misc2/_validate/stage1/lib/../lib/x86_64-linux-ghc-9.3.20211124/libHSghc-bignum-1.3-ghc9.3.20211124.so: error: undefined reference to '__gmpz_gcdext'
/home/sgraf-local/code/hs/ghc/misc2/_validate/stage1/lib/../lib/x86_64-linux-ghc-9.3.20211124/libHSghc-bignum-1.3-ghc9.3.20211124.so: error: undefined reference to '__gmpn_tdiv_qr'
/home/sgraf-local/code/hs/ghc/misc2/_validate/stage1/lib/../lib/x86_64-linux-ghc-9.3.20211124/libHSghc-bignum-1.3-ghc9.3.20211124.so: error: undefined reference to '__gmpz_sizeinbase'
/home/sgraf-local/code/hs/ghc/misc2/_validate/stage1/lib/../lib/x86_64-linux-ghc-9.3.20211124/libHSghc-bignum-1.3-ghc9.3.20211124.so: error: undefined reference to '__gmpz_export'
/home/sgraf-local/code/hs/ghc/misc2/_validate/stage1/lib/../lib/x86_64-linux-ghc-9.3.20211124/libHSghc-bignum-1.3-ghc9.3.20211124.so: error: undefined reference to '__gmpz_probab_prime_p'
/home/sgraf-local/code/hs/ghc/misc2/_validate/stage1/lib/../lib/x86_64-linux-ghc-9.3.20211124/libHSghc-bignum-1.3-ghc9.3.20211124.so: error: undefined reference to '__gmpz_nextprime'
/home/sgraf-local/code/hs/ghc/misc2/_validate/stage1/lib/../lib/x86_64-linux-ghc-9.3.20211124/libHSghc-bignum-1.3-ghc9.3.20211124.so: error: undefined reference to '__gmpz_powm'
/home/sgraf-local/code/hs/ghc/misc2/_validate/stage1/lib/../lib/x86_64-linux-ghc-9.3.20211124/libHSghc-bignum-1.3-ghc9.3.20211124.so: error: undefined reference to '__gmpz_powm_sec'
/home/sgraf-local/code/hs/ghc/misc2/_validate/stage1/lib/../lib/x86_64-linux-ghc-9.3.20211124/libHSghc-bignum-1.3-ghc9.3.20211124.so: error: undefined reference to '__gmpz_invert'
/home/sgraf-local/code/hs/ghc/misc2/_validate/stage1/lib/../lib/x86_64-linux-ghc-9.3.20211124/libHSghc-bignum-1.3-ghc9.3.20211124.so: error: undefined reference to '__gmpn_and_n'
/home/sgraf-local/code/hs/ghc/misc2/_validate/stage1/lib/../lib/x86_64-linux-ghc-9.3.20211124/libHSghc-bignum-1.3-ghc9.3.20211124.so: error: undefined reference to '__gmpn_andn_n'
/home/sgraf-local/code/hs/ghc/misc2/_validate/stage1/lib/../lib/x86_64-linux-ghc-9.3.20211124/libHSghc-bignum-1.3-ghc9.3.20211124.so: error: undefined reference to '__gmpn_ior_n'
/home/sgraf-local/code/hs/ghc/misc2/_validate/stage1/lib/../lib/x86_64-linux-ghc-9.3.20211124/libHSghc-bignum-1.3-ghc9.3.20211124.so: error: undefined reference to '__gmpn_xor_n'
/home/sgraf-local/code/hs/ghc/misc2/_validate/stage1/lib/../lib/x86_64-linux-ghc-9.3.20211124/libHSghc-bignum-1.3-ghc9.3.20211124.so: error: undefined reference to '__gmpn_mul'
/home/sgraf-local/code/hs/ghc/misc2/_validate/stage1/lib/../lib/x86_64-linux-ghc-9.3.20211124/libHSghc-bignum-1.3-ghc9.3.20211124.so: error: undefined reference to '__gmpn_cmp'
/home/sgraf-local/code/hs/ghc/misc2/_validate/stage1/lib/../lib/x86_64-linux-ghc-9.3.20211124/libHSghc-bignum-1.3-ghc9.3.20211124.so: error: undefined reference to '__gmpn_sub'
/home/sgraf-local/code/hs/ghc/misc2/_validate/stage1/lib/../lib/x86_64-linux-ghc-9.3.20211124/libHSghc-bignum-1.3-ghc9.3.20211124.so: error: undefined reference to '__gmpn_sub_1'
/home/sgraf-local/code/hs/ghc/misc2/_validate/stage1/lib/../lib/x86_64-linux-ghc-9.3.20211124/libHSghc-bignum-1.3-ghc9.3.20211124.so: error: undefined reference to '__gmpn_add_1'
/home/sgraf-local/code/hs/ghc/misc2/_validate/stage1/lib/../lib/x86_64-linux-ghc-9.3.20211124/libHSghc-bignum-1.3-ghc9.3.20211124.so: error: undefined reference to '__gmpn_mod_1'
/home/sgraf-local/code/hs/ghc/misc2/_validate/stage1/lib/../lib/x86_64-linux-ghc-9.3.20211124/libHSghc-bignum-1.3-ghc9.3.20211124.so: error: undefined reference to '__gmpn_add'
/home/sgraf-local/code/hs/ghc/misc2/_validate/stage1/lib/../lib/x86_64-linux-ghc-9.3.20211124/libHSghc-bignum-1.3-ghc9.3.20211124.so: error: undefined reference to '__gmpn_divrem_1'
/home/sgraf-local/code/hs/ghc/misc2/_validate/stage1/lib/../lib/x86_64-linux-ghc-9.3.20211124/libHSghc-bignum-1.3-ghc9.3.20211124.so: error: undefined reference to '__gmpn_mul_1'
/home/sgraf-local/code/hs/ghc/misc2/_validate/stage1/lib/../lib/x86_64-linux-ghc-9.3.20211124/libHSbase-4.16.0.0-ghc9.3.20211124.so: error: undefined reference to '__gmpn_popcount'
collect2: error: ld returned 1 exit status
`cc' failed in phase `Linker'. (Exit code: 1)
Command failed
```

in a hadrian build. These errors happen when using the stage1 compiler to compile (and link) `hpc`.

Conversely, if I don't pass `$CONFIGURE_ARGS`, then the build works as expected, but I can't link any program compiled by the stage2 compiler if I leave the nix-shell.

I'm using ghc.nix through `lorri`+`zsh` on Ubuntu 20.04.

--- a/README.md
+++ b/README.md
@@ -31,12 +31,16 @@ the compiler to be used even outsite of `nix-shell`. For convenience, the
 `nix-shell` environment also exports a convenience command, `configure_ghc`,
 which invokes `configure` as indicated.
 
+If you are using zsh, you must pass `${=CONFIGURE_ARGS}` instead; otherwise 
+zsh will escape the spaces in `$CONFIGURE_ARGS` and interpret it as one single 
+argument. See also https://unix.stackexchange.com/a/19533/61132.
+
 You can alternatively use Hadrian to build GHC:
 
 ``` sh
 $ nix-shell ~/ghc.nix/
 # from the nix shell:
-$ ./boot && ./configure $CONFIGURE_ARGS
+$ ./boot && ./configure $CONFIGURE_ARGS # In zsh, use ${=CONFIGURE_ARGS}
 # example hadrian command: use 4 cores, build a 'quickest' flavoured GHC
 # and place all the build artifacts under ./_mybuild/.
 $ hadrian/build -j4 --flavour=quickest --build-root=_mybuild

